tests/test_st10_controller.py

```diff
@@ -0,0 +1,136 @@
+"""Tests for the ST10Controller class."""
+from contextlib import nullcontext as does_not_raise
+from itertools import chain
+from typing import Any
+from unittest.mock import MagicMock, patch
+
+import pytest
+from serial import SerialTimeoutException
+
+from finesse.hardware.st10_controller import ST10Controller, ST10ControllerError
+
+
+@pytest.fixture
+def dev() -> ST10Controller:
+    """A fixture providing an ST10Controller with a patched Serial object."""
+    serial = MagicMock()
+
+    # check_device_id() and home() should both be called by ST10Controller.__init__(),
+    # but patch them for now as we'll test their inner workings elsewhere
+    with patch.object(ST10Controller, "_check_device_id") as check_mock:
+        with patch.object(ST10Controller, "home") as home_mock:
+            st10 = ST10Controller(serial)
+            check_mock.assert_called_once()
+            home_mock.assert_called_once()
```

Comment:
Why are you asserting things in a fixture?

Reply:
I have to mock them because I don't actually want these methods to run for the fixture, but I would like to check that they actually run. I suppose the danger is that if they fail they'll break all the other tests. I'll put them in a separate test